src/wallet/rpcdump.cpp

```diff
@@ -175,31 +175,29 @@ UniValue importprivkey(const JSONRPCRequest& request)
         CPubKey pubkey = key.GetPubKey();
         assert(key.VerifyPubKey(pubkey));
         CKeyID vchAddress = pubkey.GetID();
+
         {
             pwallet->MarkDirty();
 
+            // Use timestamp of 1 to scan the whole chain
+            if (!pwallet->ImportPrivKeys({{vchAddress, key}}, 1)) {
+                throw JSONRPCError(RPC_WALLET_ERROR, "Error adding key to wallet");
+            }
+
             // We don't know which corresponding address will be used;
             // label all new addresses, and label existing addresses if a
             // label was passed.
+            std::set<CScript> script_pub_keys;
             for (const auto& dest : GetAllDestinationsForKey(pubkey)) {
-                if (!request.params[1].isNull() || pwallet->mapAddressBook.count(dest) == 0) {
-                    pwallet->SetAddressBook(dest, strLabel, "receive");
```

Comment:
It's not obvious to me how dropping `SetAddressBook` doesn't change behavior, e.g. we're no longer calling `NotifyAddressBookChanged`.

Reply:
`SetAddressBook` happens in `ImportScriptPubKeys`